Lombiq.Privacy.Samples/Scripting/WorkflowHelperMethodProvider.cs

```diff
@@ -0,0 +1,42 @@
+using Microsoft.Extensions.DependencyInjection;
+using OrchardCore.Scripting;
+using OrchardCore.Workflows.Http.Controllers;
+using OrchardCore.Workflows.Http.Models;
+using OrchardCore.Workflows.Services;
+using System;
+using System.Collections.Generic;
+
+namespace Lombiq.Privacy.Samples.Scripting;
+
+public class WorkflowHelperMethodProvider : IGlobalMethodProvider
+{
+    private readonly GlobalMethod _workflowHttpEventUrlResolve;
+
+    public WorkflowHelperMethodProvider() =>
+        _workflowHttpEventUrlResolve = new GlobalMethod
+        {
+            Name = "workflowHttpEventUrlResolve",
+            Method = serviceProvider => (Func<string, string, int, object>)((workflowTypeId, activityId, tokenLifeSpan) =>
+            {
+                var securityTokenService = serviceProvider.GetRequiredService<ISecurityTokenService>();
+                var workflowTypeStore = serviceProvider.GetRequiredService<IWorkflowTypeStore>();
+
+                var workflowType = workflowTypeStore.GetAsync(workflowTypeId).GetAwaiter().GetResult();
+                var token = securityTokenService.CreateToken(
+                    new WorkflowPayload(workflowType.WorkflowTypeId, activityId),
+                    TimeSpan.FromDays(
+                        tokenLifeSpan == 0 ? HttpWorkflowController.NoExpiryTokenLifespan : tokenLifeSpan));
+
+                // LinkGenerator.GetPathByAction(...) and UrlHelper.Action(...) not resolves url for
+                // HttpWorkflowController.Invoke action.
+                // Any ideas are welcome.
```

Comment:
This is basically a TODO comment, just without the magic word. What did you try exactly? If we can't find a solution now, please open an issue in OC and include its link here.

Reply:
You can find two diffs below. One for `UrlHelper `and one for `LinkGenerator`.
The result is the same in both cases, the `url == null` at the end.

[WorkflowHelperMethodProvider_LinkGenerator.cs.diff.txt](https://github.com/Lombiq/Orchard-Privacy/files/8766487/WorkflowHelperMethodProvider_LinkGenerator.cs.diff.txt)
[WorkflowHelperMethodProvider_UrlHelper.cs.diff.txt](https://github.com/Lombiq/Orchard-Privacy/files/8766488/WorkflowHelperMethodProvider_UrlHelper.cs.diff.txt)

Reply:
Yeah, that should've worked. Please open an [OC issue](https://github.com/OrchardCMS/OrchardCore/issues) and paste the link instead of the "Any ideas are welcome." text.

Reply:
I tried both cases with root ServiceProvider too.

I started [here](https://github.com/OrchardCMS/OrchardCore/blob/62eed89890b6b9b6d71a02e1465269b9bb512207/src/OrchardCore.Modules/OrchardCore.Workflows/Http/Controllers/HttpWorkflowController.cs#L71).
The `Controller.Url` is a `IUrlHelper` instance.